src/cogent3/app/data_store_new.py

```diff
@@ -502,15 +506,23 @@ def write_log(self, *, unique_id: str, data: str) -> None:
 def get_data_source(data) -> str:
     source = getattr(data, "source", None)
     if source is None:
```

Comment:
I think our new datastores doesn't have `source` attribute. Why we use this attribute name here?

Reply:
this is for data objects, it's what the unique ID is derived from